libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,631 @@
+})
+
+
+export class ListComponent extends BaseInput implements OnInit, AfterViewInit, AfterContentInit, OnDestroy {
+
+    /**Enables lazy loadMore of data */
+    @Input()
+    loadMore: boolean;
+
+    /**Enables data load on scroll for true
+     * false: enables data loading on button
+     * click
+     */
+    @Input()
+    loadOnScroll: boolean;
+
+    /**Wait time for new items */
+    @Input()
+    delayTime: number;
+
+    /**Max height for viewport to display scroll */
+    @Input()
+    maxHeight: string;
+
+    /**Items to be loaded at once */
+    @Input()
+    itemSize: number;
+
+    /**Scroll offset percentage to trigger scroll event */
+    @Input()
+    scrollOffsetPercentage: number;
+
+    /** Whether list component has removed borders */
+    @Input()
+    noBorder = false;
+
+    @Input()
+    selection = false;
+
+    /** Whether list component has removed bottom borders */
+    @Input()
+    noSeperator: boolean;
+
+
+    /** The type of the selection. Types include:
+    *''| 'multi' | 'single'|'delete'.
+    * Leave empty for default ().'
+    * Default value is set to ''
+    */
+    @Input()
+    public selectionMode: SelectionType = '';
+
+    /**ListType 'inactive' | 'active' | 'navigation' | 'detail' */
+    @Input()
+    listType: ListType;
+
+    /**  An array that holds a list of all selected items**/
+    @Input()
+    protected selectedItems: BaseListItem[];
+
+    /** define size of items for screen reader */
+    @Input()
+    @HostBinding('attr.aria-setsize')
+    ariaSetsize: number;
+
+    /** Defines whether items are multiseletable for screen reader */
+    @Input()
+    @HostBinding('attr.aria-multiselectable')
+    ariaMultiselectable: boolean;
+
+    /** define label of list for screen reader */
+    @Input()
+    @HostBinding('attr.aria-label')
+    ariaLabel: string;
+
+    /**Title used on button when data loads on button click */
+    @Input()
+    loadTitle: string;
+
+    /**
+   * Todo: Name of the entity for which DataProvider will be loaded.
+   *  You can either pass list of
+   * items or use this entityClass and internally we should be able
+   * to do lookup to some registry
+   * and retrieve the best matching DataProvider that is set on application level
+   *
+   *
+   */
+    @Input()
+    entityClass: string;
+
+    /**
+      * Child items of the List.
+      */
+    @ContentChildren(BaseListItem, { descendants: true }) ListItems: QueryList<BaseListItem>;
+
+    @ContentChild(ListItemDef) listItemDef: ListItemDef;
+
+    /** Access child element, for checking link content*/
+    @ViewChild('link', { read: ElementRef })
+    anchor: ElementRef;
+
+    /**@hidden
+     * keyManger to handle keybord events
+     */
+    keyManager: FocusKeyManager<BaseListItem>;
+
+    /**@hidden
+     * Whether Navigation mode is included to list component
+     * for all the items
+    */
+    _navigated: boolean;
```

Comment:
`_` denotes private property so it should be private, unless its like a private for outside worlds but since you are using this in the template it needs to be public?

Reply:
updated